stringtemplate4/src/main/java/org/jdbi/v3/stringtemplate4/StringTemplateSqlLocator.java

```diff
@@ -130,21 +131,23 @@ public static STGroup findStringTemplateGroup(String path) {
      * @param path the resource path on the classpath.
      * @return the loaded StringTemplateGroup.
      */
-    public static STGroup findStringTemplateGroup(ClassLoader classLoader, String path) {
+    public static STGroup findStringTemplateGroup(ClassLoader classLoader, String path, String templateName) {
         final STGroup cached = CACHE.get(path);
         if (cached != null) {
             return cached;
         }
-        synchronized (StringTemplateSqlLocator.class) {
-            return CACHE.computeIfAbsent(path, p -> readStringTemplateGroup(classLoader, path));
+        synchronized (SYNC_OBJECT) {
+            return CACHE.computeIfAbsent(path, p -> readStringTemplateGroup(classLoader, path, templateName));
         }
     }
 
-    private static STGroup readStringTemplateGroup(ClassLoader classLoader, String path) {
+    private static STGroup readStringTemplateGroup(ClassLoader classLoader, String path, String templateName) {
         try {
             URL resource = classLoader.getResource(path);
             STGroupFile group = new STGroupFile(resource, "UTF-8", '<', '>');
             group.load();
+            // FIXME Next line will throw errors in console if variables is not defined
```

Comment:
but it will not fail. it will merely not render the attributes referenced to in the tempalte